Assets/Scripts/PDCodeGeneration/PDCodeBead.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using UnityEngine;
+
+namespace PDCodeGeneration
+{
+    public class PDCodeBead
+    {
+        public Vector3 position;
+        public readonly int componentIndex;
+        public int strand;
+        
+        private readonly int _order;
+        
+        public PDCodeBead(Bead bead, int componentIndex, int order)
+        {
+            this.componentIndex = componentIndex;
+            position = bead.position;
+            _order = order;
+        }
+        
+        public bool IsBeadAdjacent(PDCodeBead other, int numBeadsInThisComponent)
+        {
+            if (componentIndex != other.componentIndex) return false;
```

Comment:
I don't understand why this is necessary. are the next three cases and a default case not enough to determine if two beads are adjacent?

Reply:
It may be the case that the last bead in one component and the first bead in another component are checked here in which case this is necessary.